Makefile

```diff
@@ -2147,6 +2147,15 @@ GIT-USER-AGENT: FORCE
 		echo '$(GIT_USER_AGENT_SQ)' >GIT-USER-AGENT; \
 	fi
 
+GIT_BUILT_FROM_COMMIT = $(shell \
+	GIT_CEILING_DIRECTORIES="$(CURDIR)/.." \
+	git rev-parse -q --verify HEAD 2>/dev/null)
+GIT-BUILT-FROM-COMMIT: FORCE
+	@if test x'$(GIT_BUILT_FROM_COMMIT)' != x"`cat GIT-BUILT-FROM-COMMIT 2>/dev/null`" ; then \
```

Comment:
Since we already have a shell script snippet here, how about moving the assignment of `GIT_BUILT_FROM_COMMIT` into the `GIT-BUILT-FROM-COMMIT` rule? Something like:

```
@GIT_BUILT_FROM_COMMIT="$(GIT_CEILING_DIRECTORIES="$(CURDIR)/.." \
    git rev-parse -q --verify HEAD 2>/dev/null)" && \
if test x"$GIT_BUILT_FROM_COMMIT" != x"`cat GIT-BUILT-FROM-COMMIT 2>/dev/null`" ; then \
[...]
```

Reply:
Would this set `GIT_BUILT_FROM_COMMIT` as a Makefile variable, or would it only exist local to that target?

Reply:
That would be a shell variable, local to this target.